### PR TITLE
Unit Testing Framework

### PR DESCRIPTION
**Pull request by BenjaFriend**
## Feature/Issue

Created a nice little unit testing base for others to use, and that tests some of the concurrent queue functionality! 

## Implementation/Solution

For how to implement unit tests, see the `unittesting.md` file
 
## Screenshots/GIF (if applicable)
image

## Tests
 - [X] Have you reviewed your own code?
 - [X] Does it pass unit tests? _I am the unit tests_

### CODE REVIEW

File: LightVox_UnitTests/stdafx.cpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// stdafx.cpp : source file that includes just the standard includes
+// LightVox_UnitTests.pch will be the pre-compiled header
+// stdafx.obj will contain the pre-compiled type information
+
+#include "stdafx.h"
```
Comment (ttanatb) [inline LightVox_UnitTests/stdafx.cpp]: Could you explain why we need the .cpp file here?

File: LightVox_UnitTests/unittest1.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#include "stdafx.h"
```
Comment (ttanatb) [inline LightVox_UnitTests/unittest1.cpp]: Might wanna have a better name for the file

File: LightVox_UnitTests/unittest1.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+            TestQueue.emplace_front( 10 );
+            TestQueue.emplace_front( 15 );
+
+            Assert::AreEqual( TestQueue.size(), addedVals );
+
+        }
+
+	};
+}
```
Comment (ttanatb) [inline LightVox_UnitTests/unittest1.cpp]: git wants you to add another line here

**Review by ttanatb — COMMENTED** [review]
Looks mostly good! Good job @engine-buddies/tools-pod

File: LightVox_UnitTests/stdafx.cpp (hunk 1 above)
Comment (BenjaFriend) [reply LightVox_UnitTests/stdafx.cpp]: we dont

File: LightVox_UnitTests/unittest1.cpp (hunk 3 above)
Comment (BenjaFriend) [reply LightVox_UnitTests/unittest1.cpp]: Alright `CTRL +K + D`

File: LightVox_UnitTests/unittest1.cpp (hunk 2 above)
Comment (BenjaFriend) [reply LightVox_UnitTests/unittest1.cpp]: renamed to `ConcurrentQueueTest.cpp`

File: Light Vox Engine/UnitTest_1.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#include "CppUnitTest.h"
```
Comment (ttanatb) [inline Light Vox Engine/UnitTest_1.cpp]: Do we still need this file?

Comment (BenjaFriend) [reply Light Vox Engine/UnitTest_1.cpp]: What file

**Review by ttanatb — APPROVED** [review]

**Pull request merged**